packages/shared-ux/chrome/serverless_projects_documentation.md

```diff
@@ -0,0 +1,497 @@
+
+```ts
+import { type NavigationTreeDefinition } from '@kbn/shared-ux-chrome-navigation';
+
+const navigationTree: NavigationTreeDefinition = {
+  body: [
+    {
+      title: `Dashboards`,
+      link: 'dashboards',
+      getIsActive: ({ pathNameSerialized, prepend }) => {
+        return pathNameSerialized.startsWith(prepend('/app/dashboards'));
+      },
+    },
+  ],
+};
+```
+
+Technically, multiple paths can be active simultaneously and multiple nodes can be highlighted in the side navigation. However, this is not recommended as it can be confusing for the user.
+
+You can programmatically access the active paths using the `getActiveNavigationNodes$` method:
+
+```ts
+const activesNodes$ = pluginsStart.serverless.getActiveNavigationNodes$();
+```
+
+## Breadcrumbs
+
+Breadcrumbs is a list of links that represent the current navigation path. Project navigation breadcrumbs are automatically generated based on the navigation tree set for the [side navigation](#left-side-navigation) and the currently active navigation path.
+
+Project breadcrumbs are built from 3 sections:
+
+🏠 > Navigation > Deeper Context
+
+- 🏠 > Navigation - is platform controlled automatically
+- Deeper Context - is manually controlled by projects
+
+### Project breadcrumbs sections
+
+#### 🏠 Home breadcrumb
+
+Home breadcrumb is always displayed as the first breadcrumb and leads to what project specifies as home page in `kibana.yml`.
+
+#### Navigation breadcrumbs
+
+Navigation breadcrumbs are automatically generated based on the navigation tree set for the [side navigation](#left-side-navigation) and the currently active navigation path.
+
+#### Deeper context breadcrumbs
+
+Deeper context breadcrumbs are manually controlled by projects. They are added to the breadcrumbs using the `pluginsStart.serverless.setBreadcrumbs` method.
+
+```ts
+pluginsStart.serverless.setBreadcrumbs([
+  {
+    text: 'Deeper Context',
+    href: '/some/deeper/context',
+  },
+]);
+```
+
+These breadcrumbs are removed when the active navigation path changes.
+
+### Overriding breadcrumbs
+
+Projects can override navigation breadcrumbs. This will override navigation breadcrumbs, but will keep the Home breadcrumb:
+
+```ts
+pluginsStart.serverless.setBreadcrumbs([b1, b2, b3], { absolute: true });
+```
+
+The override is automatically reset when the active navigation path changes.
+
+## Toolbar
+
+The application toolbar is a horizontal bar that is displayed at the top of the page below the header where apps can put their custom actions.
+There is no serverless specific API or behavior changes for this: to set custom actions use the same API as in the default Kibana `setHeaderActionMenu` [source](https://github.com/elastic/kibana/blob/188009f9376d5976b5db37e04c52a2b5bdaf9771/packages/core/application/core-application-browser/src/app_mount.ts#L176-L205).
+There might be styling and positioning changes comparing to the toolbar in the default Kibana navigation.
```

Comment:
```suggestion
## Header Action Menu

The Header Action Menu, aka application toolbar, is a horizontal bar that is **conditionally** displayed at the top of the page below the header where apps can put their custom actions. Like the header, the toolbar has fixed position. Unlike the header however, this bar only appears when an application has registered content to show. Another UI difference, is the toolbar could be covered by flyouts or full-page overlays on the page. There might be other slight styling and positioning changes comparing to the toolbar in the default Kibana navigation.

There is no serverless specific API or behavior change for the toolbar. To set custom content in this area, use the same API as in the default Kibana `setHeaderActionMenu` [source](https://github.com/elastic/kibana/blob/188009f9376d5976b5db37e04c52a2b5bdaf9771/packages/core/application/core-application-browser/src/app_mount.ts#L176-L205). To unset content, call `setHeaderActionMenu` with `undefined`.

> NOTE: The display of the toolbar container is conditional, based on whether there is toolbar content to show. Make sure to pass `undefined` to `setHeaderActionMenu` if there is no application content to show. In classic layout, passing an empty `span` or `div` element suffices to "clear" the toolbar, but in serverless projects, it will cause a large empty container to show below the header.
```